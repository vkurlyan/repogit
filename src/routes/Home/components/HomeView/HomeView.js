import React, {PropTypes} from 'react'
import RepoList from '../RepoList/RepoList'
import './HomeView.scss'
import { PageHeader, Button, FormGroup, ControlLabel, FormControl } from 'react-bootstrap';

class HomeView extends React.PureComponent {

    constructor(state, props) {
        super(state, props);
        this.onSubmitUsername = this.onSubmitUsername.bind(this);
    }

    onSubmitUsername(e){
        e.preventDefault();
        this.props.showRepoList(this.usernameInput.value);
    }

    render() {
        const  {validationError, serverError, message, repositories} = this.props;

        return (
            <div>
                <PageHeader>List of a GitHub repositories</PageHeader>
                <form className="home__form" onSubmit={this.onSubmitUsername}>
                    <FormGroup
                        controlId="formBasicText"
                        validationState={validationError ? 'error' : null}
                    >
                        <ControlLabel>Enter a GitHub username</ControlLabel>
                        <FormControl
                            type="text"
                            placeholder="Username"
                            inputRef={(input) => { this.usernameInput = input; }}
                        />
                        {validationError && <div className="home__error">{validationError}</div>}
                    </FormGroup>

                    <Button type="submit" bsStyle="primary" >Submit</Button>

                    {serverError && <div className="home__error">{serverError}</div>}
                    {message && <div className="home__message">{message}</div>}
                </form>
                <RepoList repositories={repositories} />
            </div>
        )
    }
}

HomeView.propTypes = {
    showRepoList    : PropTypes.func.isRequired,
    repositories    : PropTypes.array.isRequired,
    validationError : PropTypes.string,
    serverError     : PropTypes.string,
    message         : PropTypes.string,
}

export default HomeView
