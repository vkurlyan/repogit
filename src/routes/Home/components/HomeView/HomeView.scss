.home__form {
  width:100%;
  padding-bottom:20px;
}

.home__error {
  color: red;
  margin-top:10px;
}

.home__message {
  margin-top:10px;
}
