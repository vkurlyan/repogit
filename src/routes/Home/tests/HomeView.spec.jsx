import HomeView from '../components/HomeView/HomeView'
import renderer from 'react-test-renderer';
import React from 'react';

const homeViewParams = {
    repositories: [],
    showRepoList: ()=>{}
};

describe('HomeView', () => {
    it('shows form', () => {
        const component = renderer.create(
            <HomeView {...homeViewParams} />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('shows messages', () => {
        const params = {
            ...homeViewParams,
            message: 'user has no repositories',
            validationError: 'it\'s required field',
            serverError: 'Server error'
        };
        
        const component = renderer.create(
            <HomeView {...params} />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
