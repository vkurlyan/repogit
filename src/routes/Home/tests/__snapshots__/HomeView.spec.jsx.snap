exports[`HomeView shows form 1`] = `
<div>
  <div
    className="page-header">
    <h1>
      List of a GitHub repositories
    </h1>
  </div>
  <form
    className="home__form"
    onSubmit={[Function]}>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor="formBasicText">
        Enter a GitHub username
      </label>
      <input
        className="form-control"
        id="formBasicText"
        placeholder="Username"
        type="text" />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit">
      Submit
    </button>
  </form>
  <ul
    className="list-group" />
</div>
`;

exports[`HomeView shows messages 1`] = `
<div>
  <div
    className="page-header">
    <h1>
      List of a GitHub repositories
    </h1>
  </div>
  <form
    className="home__form"
    onSubmit={[Function]}>
    <div
      className="form-group has-error">
      <label
        className="control-label"
        htmlFor="formBasicText">
        Enter a GitHub username
      </label>
      <input
        className="form-control"
        id="formBasicText"
        placeholder="Username"
        type="text" />
      <div
        className="home__error">
        it\'s required field
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit">
      Submit
    </button>
    <div
      className="home__error">
      Server error
    </div>
    <div
      className="home__message">
      user has no repositories
    </div>
  </form>
  <ul
    className="list-group" />
</div>
`;
